### DISCUSSION TITLE
Web interface issues with iPad air

### DISCUSSION THREAD

**Discussion by albar965** [albar965/littlenavmap#852] [General]
@u-an-i  and @KOKAProduktion 

You have an idea what can cause this?

https://www.avsim.com/forums/topic/615003-built-in-web-server-and-ipad-secondary-screen-issues/

Alex

**Comment by u-an-i** [comment]
Hi @albar965 ,

I've looked into this and 

1. confirm the main toolbar is not visible on i(Pad)OS < 14 and even some iPadOS 14 devices (such as iPad 8th Gen with iPadOS 14.0)
    **I'm investigating**.
2. am not sure which operation(s) on the map in the main .exe window is referred to. When I move the map inside the .exe window when not connected to a sim this is not reflected in the webbrowser, this is true but I don't remember whether it was or should be although I almost think it was and wonder why it wouldn't now.
3.  resizing the .exe window is not reflected in the web browser view, I am not aware how it can be reflected. Resizing the map inside the .exe window is also not reflected in the web browser. @albar965 was it ever? It also is not reflected in the legacy ui.
4. during resize a low resolution image is used. After resize completed the usual res image is loaded, testing this inside the debugger on iOS 12 still worked.

I report again.

**Reply by u-an-i** [reply]
Hello,

regarding 1. there are 2 issues: older i(Pad)OS Safaris size an iframe according to the size of the content rather than the defined size. For this, there are 2 workarounds, only one of which is viable (the other disables scrolling in the content but which we like on the Flight Plan for example) but which didn't want to work for the Flight Plan (for which the other workaround was excluded). The reason being connected to issue 2 I guess: I use "flex-box" styling, which is around for about only 12 years and for about 8 years in in its current final form, but which, in combination with an iframe creates glitched Styling on the around 1.5+ years old i(Pad)OS Safaris such as the width of a "div" box collapsing to 1px (in a vertical layout where it always would be 100%) resulting in basically the iframe dimensions not applying properly despite the workaround for that set. Additionally, despite being gotten visible, the toolbar buttons were not reacting to clicks under circumstances where I got the iframe dimension workaround to semi-apply, again due to some "flex-box" issues inside the old i(Pad)OS Safari I guess.
"flex-box" styling allows looks having dynamic, content-based stylings (such as varying number of buttons and possible varying height due to buttons wrapping at screen border while remaining content is taken up by other element) which is like I made the new design. I can replicate this with JS calculated dimensions but which is quite a large effort but which might have the effect of the iframe workaround actually working for the Flight Plan.

I try some wrappers inside the Flight Plan HTML late(r) today to try to get the iframe workaround working, other than this I'm out of ideas.

**Reply by u-an-i** [reply]
@albar965 see just added comment fyi

**Reply by u-an-i** [reply]
Hello @albar965 ,

I didn't get the "flex-box" design with iframe content to work on iOS 12 (and I guess it won't work until i(Pad)OS 14 on some devices and i(Pad)OS 15 on other devices from my testing on BrowserStack (for example iPad 5th Gen iPadOS 15 works but iPad 8th Gen iPadOS 14 does not but iPad Pro 2021 iPadOS 14 does).

I apologize for not being aware of it when creating the ui.

I can try to achieve the look without using "flex-box" and then try to workaround the iframe issue for older i(Pad)OS devices. Please let me know if you would like me to do so. It will be quite a challenge.

**Reply by albar965** [reply]
> 2\. When I move the map inside the .exe window when not connected to a sim this is not reflected in the webbrowser, this is true but I don't remember whether it was or should be although I almost think it was and wonder why it wouldn't now.
> 3\. resizing the .exe window is not reflected in the web browser view, I am not aware how it can be reflected. Resizing the map inside the .exe window is also not reflected in the web browser

Not sure if I understand but the web and the application use a separate MapPaintWidget instance. Resizing the window in the .exe should not affect the web interface at all. Let me know if it does. Only map settings like navaid visibility are copied to the web MapPaintWidget.

Regarding the fix: I would not use a big effort to implement your own flexbox design in JS. Maybe you can jump right to legacy mode if you detect unsupported browsers. No idea if this makes sense.

Alex

**Reply by u-an-i** [reply]
Hi :)

2 and 3 is what the original request is on avsim: an exact copy. I'm glad it wasn't because it isn't.

So the only issue remaining is 1. I'm also glad you don't see a JS flexbox as an requirement because the browser handling everything correctly is the future and i(Pad)OS 15 is supported on devices reaching back to 2015.

Your idea sounds very nice :) I'll implement such a workaround.

**Reply by u-an-i** [reply]
Implemented in https://github.com/albar965/littlenavmap/pull/855 for the new builds (having web plugins support), tested with BrowserStack on various iPads with i(Pad)OS from 12+ @albar965

**Comment by albar965** [comment]
Sorry, gimme a day more to answer. Fixing and preparing two releases now. :slightly_frowning_face:
